maskit/masks.py

```diff
@@ -244,13 +288,35 @@ def apply_mask(self, values: np.ndarray):
         """
         return values[~self.mask]
 
+    def mask_changed(self, mask: Mask, indices: np.ndarray):
+        """
+        Callback function that is used whenever one of the encapsulated masks does
+        change. In case the mask does change and adds a parameter back into the circuit,
+        the configured :py:attr:`~.default_value` is applied.
+
+        :raises NotImplementedError: In case an unimplemented mask reports change
+        """
+        if len(indices) == 0 or self.default_value is None:
+            return
+        np_indices = tuple(zip(*indices))
+        if not np.all(mask.mask[np_indices]):
+            if self.wire_mask == mask:
+                self.parameters[:, np_indices] = self.default_value
+            elif self.layer_mask == mask:
+                self.parameters[np_indices, :] = self.default_value
+            elif self.parameter_mask == mask:
+                self.parameters[np_indices] = self.default_value
```

Comment:
Since you don't define ``Mask.__eq__``, I recommend to use ``is`` instead of ``==``. This has the same behaviour for now, but is a) faster and b) robust if you ever *do* implement ``__eq__``.
```suggestion
            if self.wire_mask is mask:
                self.parameters[:, np_indices] = self.default_value
            elif self.layer_mask is mask:
                self.parameters[np_indices, :] = self.default_value
            elif self.parameter_mask is mask:
                self.parameters[np_indices] = self.default_value
```